compiler/modules/front/parseExprs.bal

```diff
@@ -0,0 +1,170 @@
+import wso2/nballerina.err;
+
+type Identifier readonly & record {|
+    string identifier;
+    err:Position pos;
+|};
+
+type LiteralType DECIMAL_NUMBER|STRING_LITERAL|BOOLEAN_LITERAL;
+
+type LHSExprNode Expr|Identifier;
+
+function parseExpr(Tokenizer tok) returns Expr|err:Syntax {
```

Comment:
Don't think this is handling precedence right.

Typically with a recursive descent parser you will need a separate function for each precedence level: parseAdditiveExpr, parseMultiplicativeExpr etc. For example, look at the code on parseTypeDesc

I suggest adding some tests to make sure you are getting both precedence and associativity correct.